client/components/FormattingBar/Controls/ControlsEquation.js

```diff
@@ -1,140 +1,104 @@
-import React, { Component } from 'react';
+/* eslint-disable react/no-danger */
+import React, { useRef, useEffect } from 'react';
 import PropTypes from 'prop-types';
-import { Button } from '@blueprintjs/core';
-import { renderLatexString } from 'utils';
-import Icon from 'components/Icon/Icon';
-
-const propTypes = {
-	attrs: PropTypes.object.isRequired,
-	updateAttrs: PropTypes.func.isRequired,
-	changeNode: PropTypes.func.isRequired,
-	selectedNode: PropTypes.object.isRequired,
-	editorChangeObject: PropTypes.object.isRequired,
-	isSmall: PropTypes.bool.isRequired,
-};
+import { useDebounce } from 'use-debounce';
```

Comment:
The `react-use` package that we already include has a [`useDebounce`](https://github.com/streamich/react-use/blob/master/docs/useDebounce.md) hook. If the interfaces are compatible, let's use that instead.

Reply:
Unfortunately they're not; the `use-debounce` version does by default what I almost always want — provides a debounced version of an input value directly — while the `react-use` version just lets you pass in a callback from which you might update a state value manually. I think the API of the former is useful enough to justify having around, though arguably we could also implement it ourselves in a homegrown hook in just a few lines of code.